extensions/blocks/donations/controls.js

```diff
@@ -1,16 +1,79 @@
 /**
  * WordPress dependencies
  */
-import { ExternalLink, PanelBody, ToggleControl } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
 import { InspectorControls } from '@wordpress/block-editor';
+import {
+	ExternalLink,
+	TextControl,
+	PanelBody,
+	SelectControl,
+	ToggleControl,
+} from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { SUPPORTED_CURRENCIES, minimumTransactionAmountForCurrency } from '../../shared/currencies';
 
 const Controls = props => {
 	const { attributes, setAttributes, products, siteSlug } = props;
-	const { monthlyPlanId, annuallyPlanId, showCustomAmount } = attributes;
+	const { currency, amounts, monthlyPlanId, annuallyPlanId, showCustomAmount } = attributes;
+	const [ editedAmounts, setEditedAmounts ] = useState( amounts );
+
+	const minAmount = minimumTransactionAmountForCurrency( currency );
+
+	const setAmount = ( amount, tier ) => {
+		let isValidAmount = true;
+		if ( ! amount ) {
+			isValidAmount = false;
+		}
+		amount = parseFloat( amount );
```

Comment:
Leaving some general reminders for us when working with money:

JavaScript is unfortunately all floating point (so we end up needing to do string->float), but for later work on the server, as a general reminder:
* use ints to the desired precision level (eg a number in the smallest unit eg cents or fractions of a cent)
* Avoid type conversions until the end of computations to avoid compounding rounding errors.